### PR TITLE
Parser (Fix): Output freeform content before void blocks

### LINKED ISSUE #9968: Parsing Issues with classic editor and php rendered blocks.
**Describe the bug**
in Gutenberg 3.8 where the classic editor doesnt show in the backend if a php rendered block is added just after the classic editor block.

It seems to be something to do with how the comments work. For example my php rendered block would be stored in the database like this:
```<!-- wp:my-blocks/block-custom /-->```

This causes the classic editor not to render, however if you change the comment to
```<!-- wp:my-blocks/block-custom --><!-- /wp:my-blocks/block-custom -->```
Everything will render in the editor correctly

Going back through the commits - it seems that it was caused by the new parser ( #8083 )

If my description you can see the conversation here in slack - https://wordpress.slack.com/archives/C02QB2JS7/p1537188435000100

**To Reproduce**
1. Add Classic editor block add some content
2. Directly after add the `latest posts` block (its a php rendered block)
3. Save the post
4. Refresh to see that the classic editor is not visible even though the content is in the database.

**Expected behavior**
Classic editor should be visible

**Additional context**
- Gutenberg 3.8

### PR DESCRIPTION
**Pull request by dmsnell**
Resolves #9968

It was noted that a classic block preceding a void block would
disappear in the editor while if that same classic block preceded
the long-form non-void representation of an empty block then things
would load as expected.

This behavior was determined to originate in the new default parser
in #8083 and the bug was that with void blocks we weren't sending
any preceding HTML soup/freeform content into the output list.

In this patch I've duplicated some code from the block-closing
function of the parser to spit out this content when a void block
is at the top-level of the document.

This bug did not appear when void blocks are nested because it's
the parent block that eats HTML soup. In the case of the top-level
void however we were immediately pushing that void block to the
output list and neglecting the freeform HTML.

I've added a few tests to verify and demonstrate this behavior.
Actually, since I wasn't sure what was wrong I wrote the tests first
to try and understand the behaviors and bugs. There are a few tests
that are thus not entirely essential but worthwhile to have in here.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows the accessibility standards. (N/A)
- [x] My code has proper inline documentation.

### CODE REVIEW

**Review by mcsf — APPROVED** [review]
Thanks for the quick fix, looks good!

File: packages/block-serialization-default-parser/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@wordpress/block-serialization-default-parser",
-	"version": "1.0.0-rc.0",
+	"version": "1.0.1",
```
Comment (youknowriad) [inline packages/block-serialization-default-parser/package.json]: I think the idea is that we don't touch the package.json ourselves, we let lerna do so at the moment of releasing. What we need to do though is to just add a not in the changelog and add an `( unreleased )` next to the version, that way the person that will publish the packages to npm will know what version to pick using lerna. 

you can check CHANGELOG files from other packages

Comment (dmsnell) [reply packages/block-serialization-default-parser/package.json]: thanks @youknowriad! I had no idea but TIL

**Comment by aduth** [comment]
Are there PHP tests [at all / for the fix intended here] ?

**Comment by dmsnell** [comment]
> Are there PHP tests [at all / for the fix intended here] ?

No. Good point.

**Pull request merged**